subprojects/internal-integ-testing/src/main/groovy/org/gradle/test/fixtures/ivy/IvyFileModule.groovy

```diff
@@ -340,9 +341,12 @@ class IvyFileModule extends AbstractModule implements IvyModule {
                 new VariantMetadata(
                     v.name,
                     v.attributes,
-                    dependencies.collect { d ->
+                    dependencies.findAll { d -> !d.optional }.collect { d ->
```

Comment:
I don't think Ivy dependencies support the 'optional' attribute.

Reply:
The fixture was confusing. `d.optional` meant: "this is a dependency constraint" (published to module metadata; not published to ivy.xml). I changed the fixture to maintain a separate list for constraints.